### DISCUSSION TITLE
Issue Starting Kanidm: "Failed to Generate Integrity Hash for wasmloader_admin.js"

### DISCUSSION THREAD

**Discussion by frateralexander** [kanidm/kanidm#3017] [General]
Hi

I am reading about Kanidm and plan to use it for Linux authentication via nsswitch. Initially, I downloaded the content from the Git repository and ran some commands to compile and install kanidmd on my host machine. However, when I try to start kanidmd using systemctl or the command "kanidmd server -c /etc/kanidm/server.toml", I receive the following error:

`Aug 29 13:31:53 ubuntu-jammy kanidmd[4741]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to generate integrity hash for wasmloader_admin.js - cancelling startup! | event_tag_id: 1 | err: "Can't find \"../web_ui/pkg/wasmloader_admin.js\" to generate file hash"

Aug 29 13:31:53 ubuntu-jammy kanidmd[4741]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to start HTTPS server -> ()

Aug 29 13:31:53 ubuntu-jammy kanidmd[4741]: 00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to start server core!`

In my configuration file, I set my domain as "localhost" and the origin as "https://localhost:8443".

Could someone help me ? What could be causing this error?

Thanks

**Comment by yaleman** [comment]
Typically this'll be the fact you didn't [set the correct build profile](https://kanidm.github.io/kanidm/master/developers/index.html?highlight=build%20profile#build-profiles) when building

**Comment by Firstyear** [comment]
Did you end up resolving this with the build profiles advice @frateralexander ?

**Comment by frateralexander** [comment]
Hi,

Sorry for the delay in responding.

Unfortunately, I was not successful when trying again.

I read the content of "set the correct build profile" online as well as the content in "book/src/developers/readme.md".

After installing the mentioned libraries (libudev-dev, libssl-dev, pkg-config, libpam0g-dev), I tried running the command KANIDM_BUILD_PROFILE=release_linux cargo build --release --bin kanidmd.

However, when attempting to run the command kanidmd server -c /etc/kanidm/server.toml with the configurations I mentioned earlier, I am still encountering the following error:

```
00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to generate integrity hash for wasmloader_admin.js - cancelling startup! | event_tag_id: 1 | err: "Can't find \"/usr/share/kanidm/ui/pkg/wasmloader_admin.js\" to generate file hash"
00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to start HTTPS server -> ()
00000000-0000-0000-0000-000000000000 ERROR    🚨 [error]: Failed to start server core!
```

I forgot to mention that my test environment is a Vagrant image of Ubuntu 22.04.4.

Could I have run the wrong command?

I would appreciate any help.

Thanks

**Reply by yaleman** [reply]
Where did you put the binary and where did you put the web UI contents? It's not just a single static binary

**Reply by Firstyear** [reply]
The build profile tells Kanidmd where to look for the resources *at runtime*. So it's expecting the wasmloader to be in `/usr/share/kanidm/ui/pkg/wasmloader_admin.js`.

Are you putting wasmloader_admin.js in `/usr/share/kanidm/ui/pkg`?

**Comment by frateralexander** [comment]
Hi, 

Thank you for your help; it was essential in identifying the problem. Upon reviewing things more carefully, I realized the issue was indeed related to the missing wasmloader_admin.js file, as you pointed out. I hadn’t placed the file in the correct directory, which was causing the error. After correcting that, I was able to proceed with the configuration.

I copied the files from ../server/web_ui/pkg/ to /usr/share/kanidm/ui/pkg/, and this allowed me to start the program successfully.

I apologize for not noticing the error earlier and for overlooking the log, which already indicated the source of the problem. I appreciate your patience and the clarity in your explanations — it was incredibly helpful!

Now I can continue trying to learn more about Kanidm and test it for Linux authentication.

Best regards,

**Reply by Firstyear** [reply]
No problem, glad you got it sorted!
